_posts/2020-03-05-apple-pay.md

```diff
@@ -0,0 +1,157 @@
+
+## Sandbox Testing
+
+In this section we will focus on step 2: testing on a real device using a Sandbox account.
+
+### Account Setup
+
+Before getting started implementing Apple Pay your developer account needs to be setup to support it.
+
+1. Log in to the Developer Account and select [Identifiers][5]. Add a new merchant ID by choosing an identifier and a description.
+1. Now select [Certificates][6]. Create a new decryption certificate and select Create. Select the merchant Id that you just created and upload a new certificate signing request generated by your mac's keychain.
+
+You will need this even if you're only testing since the payment will not go through without having one. Be mindful that you need to select a custom key pair information by selecting "Let me specify key pair information" and selecting ECC, key size 256 bits otherwise your certsigningrequest will be rejected.
+
+If you are a merchant you should be aware that letting your developer generate your merchant decryption certificate will allow him to be able to read all of the receipts coming from Apple Pay so you might want to do that in house even if you trust the developer (GDPR and such).
+
+
+1. Open your project and add a new capability -> Apple Pay selecting your newly created apple pay merchant id
+1. Create a new testing user on [Apple app store connect website][5]. If you're using an old test account go to that page anyway and make sure that apple pay is enabled since it was not for me. If it is not enabled you will need to create a new account. Set the account region to US to make sure it will work.
+
+After creating it you will have to set up 2 factor auth and make sure you're actually signed in. Go to Settings > iCloud > sign it. Force quit and reopen settings and see if you're actually signed it, if not sign it again. Repeat this process as amny times as necessary.
+
+The whole section on [Sandbox Testing][2] from Apple Website is worth a read.
+1. Log out from BOTH iCloud and AppStore and login to only iCloud with your test user
+
+[Apple][2]:
+>If you mistakenly use a sandbox tester account to sign in to a production environment, like iTunes, on your test device instead of your test environment, the sandbox account becomes invalid and can’t be used again. If this happens, create a new sandbox tester account with a new email address.
+
+1. Open Wallet.app and add test credit cards found on [apple pay sandbox website][2]
+If the credit cards do not work here are some workaround:
+- Use a US testing account and change region to US [This was the one that worked for me]
+- Try a different card from the Apple provided list. Be aware that the selected card will need to match the merchant's capabilities
+- Logout and login again on both iCloud and AppStore
+- Force quit Wallet and restart iPhone
+- Use a real credit card instead of a test one (!? Yes this was a suggested solution)
+1. Inside Wallet.app change the shipping address to a real address
```

Comment:
1. - > -

Reply:
well why it's a 1.

Reply:
markdown converts it to the correct number itself, so all good there